analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCognitiveComplexityMetric.cs

```diff
@@ -85,6 +94,19 @@ public override void Visit(SyntaxNode node)
                 }
             }
 
+            public override void VisitCompilationUnit(CompilationUnitSyntax node)
+            {
+                foreach (var globalStatement in node.Members.Where(x => x.IsKind(SyntaxKind.GlobalStatement)))
+                {
+                    base.Visit(globalStatement);
+                }
+
+                if (!onlyGlobalStatements)
```

Comment:
Instead of adding this extra bool `onlyGlobalStatements`, I would check inside the `VisitCompilationUnit` if the first symbolic member is in a top-level program with this [method](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CFG/ShimLayer/INamedTypeSymbolExtensions.cs#L34).
Unless there's some other use case for this bool that I am missing.

Reply:
As discussed with @dmitry-golovinov-sonarsource this bool is needed in order to have compatibility also with other components.